### PR TITLE
Mention pixi in installation guide

### PR DESCRIPTION
**Pull request by pavelzw**
## Overview: What does this pull request change?

Mention pixi in the installation guide

## Motivation and Explanation: Why and how do your changes improve the library?

[Pixi](https://pixi.sh) builds on top of the conda package ecosystem and solves many problems that conda has suffered from like automatic generation of lockfiles, no weird `conda activate`, being very slow, painful installation process, etc.

## Links to added or changed documentation pages

https://manimce--3678.org.readthedocs.build/en/3678/installation.html

https://manimce--3678.org.readthedocs.build/en/3678/installation/conda.html

## Reviewer Checklist
- [x] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: docs/source/installation/conda.rst [hunk 1]
```diff
@@ -4,18 +4,21 @@ Conda
...
+There are several package managers that work with conda packages,
+namely `conda <https://docs.conda.io/projects/conda/en/latest/user-guide/install/download.html>`__,
+`mamba <https://mamba.readthedocs.io>`__ and `pixi <https://pixi.sh>`__.
 
-After installing conda, you can create a new environment and install ``manim`` inside by running
+After installing your package manager, you can create a new environment and install ``manim`` inside by running
 
 .. code-block:: bash
 
+   # using conda
```
Comment (JasonGrace2282) [inline docs/source/installation/conda.rst:15]:
```suggestion
   # using conda or mamba
```

**Review by JasonGrace2282 — CHANGES_REQUESTED** [review]
Have you tested manim using `pixi`?

**Comment by JasonGrace2282** [comment]
@behackl, what do you think about officially supporting `pixi`? The newest version has a compatibility layer with conda so it shouldn't (?) be too much work?

**Comment by pavelzw** [comment]
Pixi basically is conda under the hood (at least it uses the same package ecosystem).
Everything that works in conda should work equally as good (in most cases even better) in pixi.

**Comment by behackl** [comment]
I am not quite convinced yet. The support for `pyproject.toml` manifest files in the current release of pixi [seems to be experimental](https://prefix.dev/blog/using_python_projects_with_pixi), and pixi does not seem to be particularly mature at this point.

Where is the advantage of using pixi over poetry for project management -- just the fact that it, like conda, would also play the role of `pyenv` in managing the python environment too?

**Comment by pavelzw** [comment]
> The support for pyproject.toml manifest files in the current release of pixi [seems to be experimental](https://prefix.dev/blog/using_python_projects_with_pixi)

The support for pyproject.toml manifest files is basically the same as using a regular `pixi.toml` but writing `tool.pixi` before everything.
The support for pypi dependencies is actually going out of beta right in the next release: https://github.com/prefix-dev/pixi/pull/1120

> pixi does not seem to be particularly mature at this point

From my experiences, pixi is actually pretty mature nowadays and supports most use cases that conda also supports.

> Where is the advantage of using pixi over poetry for project management

I think the main difference is the pypi world vs. conda world.

I see two main advantages of the conda world compared to the pypi world:

1. It's possible to build conda packages in a reproducible manner (i.e. in conda-forge ci build pipelines that repackage the libraries from source) instead of arbitrary wheels that random package developers upload. This also allows for easier migrations to other architectures (for example M1 macs have had better support in the conda ecosystem at the beginning because there was a central migration instead of the need to ping every developer that they should include m1 builds as well in their wheels).
2. Conda technically is a language-agnostic package manager. This allows for non-python dependencies to be installed into the environment as well. In manim's case, conda would also install ffmpeg (so you don't need to install it additionally to the env) but in general, I think this is especially helpful for the python world as python bindings to other libraries are way easier to actually build and ship with this approach: you can just depend on the original C++ libraries during the build process and not expect them to be magically there / torture your `setup.py` to include them from somewhere. This is especially helpful in the scientific computing community.

Pixi is just a _way_ better conda since it actually thinks about the question "what do people need in terms of project management?". You could say that pixi is the "poetry of the conda world".
Also see https://pixi.sh/latest/FAQ/#what-is-the-difference-with-conda-mamba-poetry-pip for a comparison
Also, pixi now properly supports pypi packages on top of conda packages which you need in some cases (where there isn't a conda package yet).

Back to the original question:
For me personally, there is actually one major annoyance I've found with poetry (which is why i have never _properly_ given it a fair shot): You can't have different features of dependencies that contradict each other, poetry always wants all dependencies to be installable at once. In some projects, you might want to test whether your project works with _both_ `pydantic 1` and `pydantic 2`. From what I've seen, poetry doesn't support this use case which I think is a very important one when developing libraries. You need multiple environments for this (which pixi supports).
(this particular feature is not really relevant when building manim videos but i just wanted to put it out there because you asked)

File: docs/source/installation.rst [hunk 2]
```diff
@@ -28,10 +28,10 @@ your system's Python, or via Docker).
 
 .. _conda-installation:
 
-Installing Manim in conda
-*************************
+Installing Manim in conda or pixi
+*********************************
```
Comment (behackl) [inline docs/source/installation.rst:32]:
```suggestion
Installing Manim via Conda and related environment managers
***********************************************************
```

File: docs/source/installation.rst [hunk 3]
```diff
@@ -28,10 +28,10 @@ your system's Python, or via Docker).
...
 
-Conda is a package manager for Python that allows creating environments
+Conda and pixi are package managers for Python that allows creating environments
```
Comment (behackl) [inline docs/source/installation.rst:34]:
```suggestion
Conda is a package manager for Python that allows creating environments
```

**Review by behackl — CHANGES_REQUESTED** [review]
Thanks for the elaborate answer @pavelzw! Just to make sure, I did not intend to sound rude earlier; apologies if it looked or felt like it.

I have a better feeling for the role pixi intends to play now. Generally speaking, I do like the idea and I think it is beneficial to mention it in our documentation as well. However, I'd propose doing so a bit less prominently (e.g., in form of a note-admonition within the conda section in `installation.rst`, without turning the section into a conda-and-pixi-section). A more concrete suggestion that could be included in `installation.rst`:

```rest
.. NOTE::

   There are various popular alternatives to Conda like
   `mamba <https://mamba.readthedocs.io/en/latest/>`__ /
   `micromamba <https://mamba.readthedocs.io/en/latest/user_guide/micromamba.html>`__,
   or `pixi <https://pixi.sh>`__.
   They all can be used to setup a suitable, isolated environment
   for your Manim projects.
```

As a consequence of #3501 and the removal of the external dependency on ffmpeg we'll need to modify (and in particular: simplify!) all of the installation guides generously anyhow.

**Comment by pavelzw** [comment]
Sounds good 👍🏻 

> Just to make sure, I did not intend to sound rude earlier; apologies if it looked or felt like it.

No offense taken :)

**Review by behackl — APPROVED** [review]
SGTM. Thanks!

**Pull request merged**